pkg/controllers/termination/v1alpha1/suite_test.go

```diff
@@ -0,0 +1,97 @@
+package v1alpha1
+
+import (
+	"context"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/Pallinder/go-randomdata"
+	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha1"
+	"github.com/awslabs/karpenter/pkg/cloudprovider"
+	"github.com/awslabs/karpenter/pkg/cloudprovider/fake"
+	"github.com/awslabs/karpenter/pkg/test"
+	webhooksprovisioning "github.com/awslabs/karpenter/pkg/webhooks/provisioning/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	. "github.com/awslabs/karpenter/pkg/test/expectations"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest/printer"
+)
+
+func TestAPIs(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecsWithDefaultAndCustomReporters(t,
+		"Terminator",
+		[]Reporter{printer.NewlineReporter{}})
+}
+
+var controller *Controller
+var env = test.NewEnvironment(func(e *test.Environment) {
+	cloudProvider := fake.NewFactory(cloudprovider.Options{})
+	controller = NewController(
+		e.Manager.GetClient(),
+		corev1.NewForConfigOrDie(e.Manager.GetConfig()),
+		cloudProvider,
+	)
+	e.Manager.RegisterWebhooks(
+		&webhooksprovisioning.Validator{CloudProvider: cloudProvider},
+		&webhooksprovisioning.Defaulter{},
+	).RegisterControllers(controller)
+})
+
+var _ = BeforeSuite(func() {
+	Expect(env.Start()).To(Succeed(), "Failed to start environment")
+})
+
+var _ = AfterSuite(func() {
+	Expect(env.Stop()).To(Succeed(), "Failed to stop environment")
+})
+
+var _ = Describe("Reallocation", func() {
+	var provisioner *v1alpha1.Provisioner
+	var ctx context.Context
+
+	BeforeEach(func() {
+		// Create Provisioner to give some time for the node to reconcile
+		provisioner = &v1alpha1.Provisioner{
+			ObjectMeta: metav1.ObjectMeta{Name: strings.ToLower(randomdata.SillyName()),
+				Namespace: "default",
+			},
+			Spec: v1alpha1.ProvisionerSpec{
+				Cluster: &v1alpha1.ClusterSpec{Name: "test-cluster", Endpoint: "http://test-cluster", CABundle: "dGVzdC1jbHVzdGVyCg=="},
+			},
+		}
+		ctx = context.Background()
```

Comment:
nit: consider moving this above provisioner to reduce eye jumping when reading code (i.e. you can read provisioner and ctx right next to each other, and know that provisioner is the last variable).